version: "3.7"

services:
  nginx:
    image: nginx:1.19.5
    networks:
      - taroun_1ban_network
    volumes:
      - taroun_1ban_static:/data/static/
      - taroun_1ban_media:/data/media/
      - /home/aksghk4671/nginx.conf:/etc/nginx/nginx.conf
    ports:
      - 80:80

  django:
    image: taroun_1ban_django:4
    networks:
      - taroun_1ban_network
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD
    volumes:
      - taroun_1ban_static:/home/new_1ban/staticfiles/
      - taroun_1ban_media:/home/new_1ban/media/

  mariadb:
    image: mariadb:10.5
    networks:
      - taroun_1ban_network
    volumes:
      - taroun_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_USER
      - MARIADB_PASSWORD
      - MARIADB_ROOT_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER: django
      MARIADB_PASSWORD: tata1212
      MARIADB_ROOT_PASSWORD: tata1212

networks:
  taroun_1ban_network:

volumes:
  taroun_1ban_static:
  taroun_1ban_media:
  taroun_mariadb:

secrets:
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  DJANGO_SECRET_KEY:
    external: true